projects/media-viewer/src/lib/annotations/comments-summary/comments-summary.component.ts

```diff
@@ -1,37 +1,93 @@
+import {FormBuilder, FormControl, FormGroup} from '@angular/forms';
+
+import * as fromTagSelectors from '../../store/selectors/tags.selectors';
 
 @Component({
   selector: 'mv-comments-summary',
   templateUrl: './comments-summary.component.html',
 })
-export class CommentsSummaryComponent implements OnInit {
+export class CommentsSummaryComponent implements OnInit, OnDestroy {
 
   @Input() title: string;
   @Input() contentType: string;
-  public comments$: Observable<{comment: string; tags: object[]; date: string; user: string; page: string}>
   @ViewChild('commentContainer') commentsTable: ElementRef;
+  public comments$: Observable<any>;
+  public filtersFg: FormGroup;
+  private $subscriptions: Subscription;
+  allTags$: Observable<{key: string; length: number}[]>;
+  showFilters = false;
+  hasFilter = false;
 
   constructor(
     private store: Store<fromStore.AnnotationSetState>,
     private readonly printService: PrintService,
     private readonly toolbarEvents: ToolbarEventService,
-    private readonly viewerEvents: ViewerEventService
+    private readonly viewerEvents: ViewerEventService,
+    private fb: FormBuilder
   ) {}
 
   ngOnInit(): void {
-    this.comments$ = this.store.pipe(select(fromSelectors.getCommentSummary), tap(console.log));
+    this.filtersFg = this.fb.group({
+      dateRangeFrom: new FormGroup({
+        day: new FormControl(''),
+        month: new FormControl(''),
+        year: new FormControl('')
+      }),
+      dateRangeTo: new FormGroup({
+        day: new FormControl(''),
+        month: new FormControl(''),
+        year: new FormControl('')
+      }),
+      tagFilters: this.fb.group({}),
+    });
+    this.comments$ = this.store.pipe(select(fromSelectors.getCommentSummary));
+    this.buildCheckBoxFrom();
+  }
+
+  buildCheckBoxFrom() {
+    this.filtersFg.reset();
+    const checkboxes = <FormGroup>this.filtersFg.get('tagFilters');
+    const filters$ = this.store.pipe(select(fromSelectors.getCommentSummaryFilters));
+    this.allTags$ = this.store.pipe(select(fromTagSelectors.getAllTagsArr));
+    this.$subscriptions = combineLatest([this.allTags$, filters$]).subscribe(([tags, filters]) => {
+      this.hasFilter = filters.hasFilter;
+      tags.forEach((val) => {
+        const checkBoxValue = (filters.filters.tagFilters &&
+          filters.filters.tagFilters.hasOwnProperty(val.key)) ?
+          filters.filters.tagFilters[val.key] : false;
+        checkboxes.addControl(val.key, new FormControl(checkBoxValue));
+      });
+      this.filtersFg.updateValueAndValidity();
+    });
+  }
+
+  onClearFilters() {
+    this.store.dispatch(new fromAnnoActions.ClearCommentSummaryFilters());
+    this.buildCheckBoxFrom();
+  }
+
+  onFilter() {
+    const {value} = this.filtersFg;
+    const hasDateFrom =  (value.dateRangeFrom.year && value.dateRangeFrom.month && value.dateRangeFrom.day);
+    const hasDateTo = (value.dateRangeTo.year && value.dateRangeTo.month && value.dateRangeTo.day);
```

Comment:
Don't need the parentheses

Reply:
it kind of reads better to me i tend always do this